app/core/calculations/groupinfo.py

```diff
@@ -40,8 +40,11 @@ class GroupInfo:
 class GroupData:
     """
     Associates Groups with their GroupInfo.
+    group is a deep copy.
     """
+    # Deep-copied.
```

Comment:
This might be an overreach but would this increment our RAM consumption ? Probably nothing to worry about, just a thought.

Reply:
Doubtful... at first I had thought that we might be able to get away with copying less, but I was wrong, so we probably could have done a straightforward deepcopy.

Nonetheless, I learned a few things... so... 😅